src/content/drafts/how-to-create-a-custom-eslint-rule.md

```diff
@@ -0,0 +1,286 @@
+
+Take some time to explore what code translates to what nodes by copying / pasting in various chunks of code that relate to the rule you're trying to make. This will make it easier for you when you go to write the code for the rule.
+
+## Generate the files
+
+It's now time to go about actually making your rule. (Yay!) Each rule in ESLint has three files named with its identifier. 
+
+*Note*: An identifier is a short, high-level description of the rule separated by dashes.
+
+For the bacon example, the identifier could be something like `no-bacon`. In which case the three files would be:
+
+1. `tests/lib/rules/no-bacon.js` -- A test file where you will test the rule to make sure it works as expected. I'd recommend you begin by writing this before you write your source file.
+2. `lib/rules/no-bacon.js` -- The source file where you will write the core logic of the rule.
+3. `docs/rules/no-bacon.md` -- A Markdown file that explains what your rule does. It largely will contain the answers to the questions discussed above.
+
+**Before examining each file, there's a few important links you should be aware of**:
+
+* ESLint provides a [Working with Rules](https://eslint.org/docs/latest/developer-guide/working-with-rules) guide on their website that goes into a lot of detail about each of the files and what every property is. It's something you'll definitely want to check out when you start making your own rules.
+* There is a [Yeoman](https://yeoman.io/) generator that aids in the creation of plugins and rules that you can find [here](https://github.com/eslint/generator-eslint). Once installed, it'll ask you a series of questions and then generate all of the files in the correct spots for you -- with much of the core code set up for you. I found it really useful in helping me get started with rules. 
+
+### Test File (tests/lib/rules/no-bacon.js)
+
+The Test File is what I'd recommend starting with. If you've followed along with the guide so far, the work you did earlier on deciding what will make your rule pass / fail can be used here. In fact the test file will consist almost entirely of those examples!
+
+To begin your test file, you'll want to import your rule and the ESLint `RuleTester`:
+
+```javascript
+const rule = require("../../../lib/rules/no-bacon");
+const RuleTester = require("eslint").RuleTester;
+```
+
+From there, you'll want to create your own `ruleTester` object that you can use to run your test cases. You can provide it with options that will act as defaults for all the test cases. 
+
+For instance, the `RuleTester`, by default, uses [ECMA Version 5](https://www.w3schools.com/js/js_es5.asp). If you're used to modern JavaScript, some of your test cases might not work. To fix that, you can specify `parserOptions` when you make this tester so that all your tests use ECMA Version 6 instead:
```

Comment:
```suggestion
For instance, the `RuleTester`, by default, uses ECMAScript 5. If you're used to modern JavaScript, some of your test cases might not work. To fix that, you can specify `parserOptions` when you make this tester so that all your tests use ECMAScript 6 instead by using the `ecmaVersion` option:
```